core/Cargo.toml

```diff
@@ -8,44 +8,58 @@ license = "MIT"
 
 [dependencies]
 anyhow = "1"
-arrayvec = "0.7.1"
 async-trait = "0.1"
-async-channel = { version = "1.6", optional = true }
 beef = { version = "0.5.1", features = ["impl_serde"] }
+futures-channel = "0.3.14"
+jsonrpsee-types = { path = "../types", version = "0.9.0" }
 thiserror = "1"
-futures-channel = { version = "0.3.14", default-features = false }
+serde = { version = "1.0", default-features = false, features = ["derive"] }
+serde_json = { version = "1", features = ["raw_value"] }
+
+# optional deps
+arrayvec = { version = "0.7.1", optional = true }
+async-channel = { version = "1.6", optional = true }
+async-lock = { version = "2.4", optional = true }
 futures-util = { version = "0.3.14", default-features = false, optional = true }
-hyper = { version = "0.14.10", default-features = false, features = ["stream"] }
-jsonrpsee-types = { path = "../types", version = "0.9.0"}
+hyper = { version = "0.14.10", default-features = false, features = ["stream"], optional = true }
 tracing = { version = "0.1", optional = true }
 rustc-hash = { version = "1", optional = true }
 rand = { version = "0.8", optional = true }
-serde = { version = "1.0", default-features = false, features = ["derive"] }
-serde_json = { version = "1", features = ["raw_value"] }
-soketto = "0.7.1"
+soketto = { version = "0.7.1", optional = true }
 parking_lot = { version = "0.12", optional = true }
 tokio = { version = "1.8", features = ["rt"], optional = true }
+wasm-bindgen-futures = { version = "0.4.19", optional = true }
+futures-timer = { version = "3", optional = true }
 
 [features]
 default = []
-http-helpers = ["futures-util"]
+http-helpers = ["hyper", "futures-util"]
 server = [
+	"arrayvec",
 	"async-channel",
-	"futures-util",
-	"rustc-hash",
+	"futures-util/alloc",
+	"rustc-hash/std",
 	"tracing",
 	"parking_lot",
 	"rand",
 	"tokio",
 ]
-client = ["futures-util"]
+client = ["futures-util/sink", "futures-channel/sink", "futures-channel/std"]
 async-client = [
+	"async-lock",
 	"client",
 	"rustc-hash",
-	"tokio/sync",
-	"tokio/macros",
-	"tokio/time",
-	"tracing"
+	"tokio",
+	"tracing",
+	"futures-timer",
+]
+async-wasm-client = [
```

Comment:
we could reduce the features the `async client` was extracted to a separate crate...